tests/unit/src/test/scala/tests/SemanticTokensLspSuite.scala

```diff
@@ -171,7 +171,7 @@ class SemanticTokensLspSuite extends BaseLspSuite("SemanticTokens") {
     "self-type",
     """|<<package>>/*keyword*/ <<a>>/*namespace*/
        |
-       |<<object>>/*keyword*/ <<Abc>>/*class*/ { <<self>>/*variable,readonly*/: <<Any>>/*class*/ <<=>>>/*operator*/
+       |<<object>>/*keyword*/ <<Abc>>/*class*/ { self: Any <<=>>>/*operator*/
```

Comment:
I feel like `Any` should be marked as `class` here.

Reply:
Looks like no features work on `Any` (actually its not specific to `Any`, but I haven't yet figured the pattern) in self type, including hover and document highlight, because there is no symbol at this node in the AST. 
I think its best to work on it in a follow up, since its not specifically related to semantic highlighting. I would still be coloured as class by the editor

Reply:
It seems to work with the `self` type in line `192`, so I though it's a strictly `Any`/`AnyRef` type of situation. Anyway let's do it as a follow up .

Reply:
fixed in https://github.com/scalameta/metals/pull/6006